### DISCUSSION TITLE
Eventual planning for dropping Disqus

### DISCUSSION THREAD

**Discussion by Starz0r** [chocolatey/home#162] [General]
This has probably been said multiple times before, but I'd like to open an issue for it if it eventually gets picked up. Let's be real, Disqus has not had the best [track record](https://en.wikipedia.org/wiki/Disqus#Criticism,_privacy,_and_security_concerns), but it's consistently awful enough that I think real consideration should be taken to moving away from it. So let's take a look at some things.

What does Disqus provide?:
- A place where comments can be placed to discuss a package. Very useful for pointing out issues or helping out with support if something is going wrong and the maintained can't be reached.
- Proper analytics to understand when users are engaging with certain packages.
- Monetization possibly? (I'm unsure if you guys are actually using this feature, but we'll leave it open).
- Still worked on. Most SaaS business come and go, even all types of software tends to, but Disqus is still around from its initial inception in 2007, which sounds like it'll probably be around for even longer!
- Cheap, and affordable. We definitely don't want to be spending a fortune on commenting software, the cheaper, the better.

Let's discuss the main issues:
- Privacy Concerns. From the outset, the company behind Disqus does not seem to care one bit about handling sensitive data, and will regularly abuse their power whenever necessary.
- Terms of Service leaves [much to be desired](https://tosdr.org/en/service/239). They seem to ensure you waive a ton of rights as long as you are using the service, which makes me think they aren't considerate to their users at all.
- Proprietary JavaScript required to use the service. This might seem like a non-issue at first, but Chocolatey is a FOSS service, and the website is mostly free of un-liberated JavaScript. Disqus seems to be the one exception in which proprietary JavaScript is required in order to interact with the service. I see this is a big disservice to those using the website, even if the majority of users are, ironically, Microsoft Windows users.

Now that we've outlined what they provide, and all their problems, let's go over what a possible replacement would need:
- Provides commenting as a baseline feature.
- Terms of Service should be respectable, while still respecting the user.
- Must absolutely be free of non-free components and not impose a restrictive license on the users.
- Well maintained.
- Possibly allows for migration from Disqus.
- Affordable.
- Service owners shouldn't be looking to abuse the privacy of users at any moment they get.
- Analytics, some monetization (?).

From my research, there were a lot of potential candidates, but the lot of them were held back from not being continuously maintained. The best one I saw that checked 7 out of the 8 boxes was [Commento](https://commento.io/). It only recently seems to slowed down in being worked on, but I imagine if they had some supporters, financial or otherwise, we might see it pick up steam again, which would be great.

Either way, I really think this is worth thinking about in the long run as Disqus has shown they're pretty much willing to abuse user trust for their own personal gain, and I do not think we should contribute to their conning of users.

**Comment by TheCakeIsNaOH** [comment]
I agree that Disqus is not ideal.

Given that the current chat service (Gitter) is now being updated to run matrix as a backend, using a chat service that can bridge/interact with gitter might open up some neat features.

I really like the concept of [Cactus Comments](https://cactus.chat/), however it is still under heavy development. It very similar to Disqus in user experience, but uses Matrix as the backend, so you can comment from any Matrix client, not just the webpage. 

There is one more requirement I would add to the requirements for a possible replacement, which is good support for moderation. From what I understand, currently all new commenters have to have their first comment approved by a site admin, and I think that would have to continue.

**Comment by pauby** [comment]
Just to manage the expectations, this is a very low priority for us and not something we will look at any time soon.

**Comment by Elaborendum** [comment]
I know it's not a priority, but I want to mention giscus.app and utteranc.es, which use GitHub Discussions and GitHub Issues respectively as backends.
